src/main/java/net/dries007/tfc/common/tileentity/TickCounterTileEntity.java

```diff
@@ -0,0 +1,52 @@
+package net.dries007.tfc.common.tileentity;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.tileentity.TileEntityType;
+
+import net.dries007.tfc.util.calendar.Calendars;
+
+public class TickCounterTileEntity extends TFCTileEntity
+{
+    private long lastUpdateTick;
+
+    public TickCounterTileEntity()
+    {
+        this(TFCTileEntities.TICK_COUNTER.get());
+    }
+
+    protected TickCounterTileEntity(TileEntityType<?> type)
+    {
+        super(type);
+        resetCounter();
+    }
+
+    public long getTicksSinceUpdate()
+    {
+        return Calendars.SERVER.getTicks() - lastUpdateTick;
+    }
+
+    public void resetCounter()
+    {
+        lastUpdateTick = Calendars.SERVER.getTicks();
```

Comment:
Calling markDirty here actually breaks stuff. Because for some reason the TE is not fully created if you place it and then call resetCounter() right afterwards. It will send 1 invalid packet before becoming normal again. I will have to talk with you about this at some point